test.ts

```diff
@@ -4,45 +4,57 @@ import * as path from 'path';
 import * as fs from 'fs';
 import { randomBytes } from 'node:crypto';
 import { decompress } from '@mongodb-js/zstd';
-
-// This number is used to start the distribution of ports to all the containers that are being run
-const PORT_BASE = 8000;
+import { TestCase } from './testCase.spec';
 
 // A salt for docker compose project names such that they do not conflict with previous or parallel test runs
 const docker_project_salt = randomBytes(Math.ceil(2)).toString('hex').slice(0, 4);
 
-function getTestsetDirectories() {
-    const testsetsPath = path.resolve(__dirname, 'testsets');
-    return fs
-        .readdirSync(testsetsPath)
-        .filter((name) => fs.lstatSync(path.join(testsetsPath, name)).isDirectory())
-        .map((name) => path.join(testsetsPath, name));
-}
+type TestSuite = {
+    path: string;
+    lapisPort: number;
+    siloPort: number;
+    testCases: TestCase[];
+};
 
-const testsets: [string, number, number][] = getTestsetDirectories().map((dir, index) => {
-    const port1 = PORT_BASE + index * 2;
-    const port2 = PORT_BASE + index * 2 + 1;
-    return [dir, port1, port2];
-});
+const PORT_BASE = 8000;
+
+const testsets: TestSuite[] = await Promise.all(
+    getTestsetDirectories().map(async (dir, index): Promise<TestSuite> => {
+        const port1 = PORT_BASE + index * 2;
+        const port2 = PORT_BASE + index * 2 + 1;
+
+        const queriesDir = path.join(dir, 'queries');
+
+        const testCases: TestCase[] = await Promise.all(
+            fs
+                .readdirSync(queriesDir)
+                .filter((x) => x.endsWith('.query.ts'))
+                .map(async (file) => await loadTestObject(path.join(queriesDir, file))),
+        );
+
+        return { path: dir, lapisPort: port1, siloPort: port2, testCases: testCases };
```

Comment:
You can use the shorthand notation if the key and value names are the same.

```suggestion
        return { path: dir, lapisPort, siloPort, testCases };
```